File: python/darts/darts.py
import math

SCORES = {1: 10, 5: 5, 10: 1}


def score(x: int, y: int) -> int:
    diag = math.sqrt(x*x + y*y)
    return next(
        (
            score for distance, score in SCORES.items()
            if diag <= distance
        ),
        0,
    )

File: go/darts/darts.go
package darts

import "math"

type scoreLine struct {
	threshold float64
	points    int
}

var scores = []scoreLine{
	{10, 0},
	{5, 1},
	{1, 5},
	{-1, 10},
}

func Score(x, y float64) int {
	distance := math.Sqrt(math.Pow(x, 2) + math.Pow(y, 2))
	for _, line := range scores {
		if distance > line.threshold {
			return line.points
		}
	}
	panic("How did you get here?")
}
